pkg/operators/wasm/maps.go

```diff
@@ -236,7 +255,31 @@ func (i *wasmOperatorInstance) mapUpdate(ctx context.Context, m wapi.Module, sta
 		return
 	}
 
-	err = ebpfMap.Update(key, value, ebpfFlags)
+	if getFD {
+		if !mapAcceptsFD(ebpfMap) {
+			i.logger.Warnf("mapUpdate: map type %v does not accept FD, expecting one in: %v", ebpfMap.Type(), mapTypesAcceptingFD)
+			stack[0] = 1
+			return
+		}
+
+		targetMapHandle := binary.NativeEndian.Uint32(value)
+		targetMap, ok := getHandle[*ebpf.Map](i, targetMapHandle)
+		if !ok {
+			stack[0] = 1
+			return
+		}
+
+		err = ebpfMap.Update(key, uint32(targetMap.FD()), ebpfFlags)
+	} else {
+		if mapAcceptsFD(ebpfMap) {
+			i.logger.Warnf("mapUpdate: map type %v accepts FD, expecting any map except one in: %v", ebpfMap.Type(), mapTypesAcceptingFD)
+			stack[0] = 1
+			return
+		}
+
+		err = ebpfMap.Update(key, value, ebpfFlags)
+	}
```

Comment:
At this point I'm really wondering if we need the getFD flag. Can't we just imply it from the map type? If trying to update a "mapAcceptsFD" map, then treat the value as a handle. I know I suggested the flag before, but if we're going to implement it in this way, I think we can avoid it. 

Is there anything I am missing?